File: htmltext/src/main/java/com/wt/htmltext/span/ImageClickSpan.java
package com.wt.htmltext.span;

import android.content.Context;
import android.text.TextPaint;
import android.text.style.ClickableSpan;
import android.view.View;
import com.wt.htmltext.OnTagClickListener;

import java.util.List;


public class ImageClickSpan extends ClickableSpan {
    private OnTagClickListener listener;
    private final Context context;
    private final List<String> imageUrls;
    private final int position;

    public ImageClickSpan(Context context, List<String> imageUrls, int position) {
        this.context = context;
        this.imageUrls = imageUrls;
        this.position = position;
    }

    public void setListener(OnTagClickListener listener) {
        this.listener = listener;
    }

    @Override
    public void onClick(View widget) {
        if (listener != null) {
            listener.onImageClick(context, imageUrls, position);
        }
    }

    @Override
    public void updateDrawState(TextPaint ds) {
        ds.setColor(ds.linkColor);
        ds.setUnderlineText(false);
    }
}

File: htmltext/src/main/java/com/wt/htmltext/span/ImageClickSpan.kt
package com.wt.htmltext.span

import android.content.Context
import android.text.style.ClickableSpan
import com.wt.htmltext.OnTagClickListener
import android.text.TextPaint
import android.view.View

class ImageClickSpan(
    private val context: Context,
    private val imageUrls: List<String?>,
    private val position: Int
) : ClickableSpan() {
    private var listener: OnTagClickListener? = null
    fun setListener(listener: OnTagClickListener?) {
        this.listener = listener
    }

    override fun onClick(widget: View) {
        listener?.onImageClick(context, imageUrls, position)
    }

    override fun updateDrawState(ds: TextPaint) {
        ds.color = ds.linkColor
        ds.isUnderlineText = false
    }
}